### DISCUSSION TITLE
Running Phoss SMP with Java 11

### DISCUSSION THREAD

**Discussion by OysteinLq** [phax/phoss-smp#200] [General]
Hi, we've been running Phoss SMP with success using Wildfly for a few years, but have recently made an attempt at upping from Java 8 to Java 12 and found that we get a steady stream of JAXB-related errors in the log.
We cannot see anything in the documentation about which versions of Java Phoss supports or recommends, so thought I'd ask.

The service itself seems o work perfectly fine despite these errors, but naturally we'd like to figure out what we might be doing wrong.

` [com.helger.jaxb.GenericJAXBMarshaller] (default task-10) Failed to set the namespace context [MapBasedNamespaceContext@0x270e995f: prefix2NS={smp=http://busdox.org/serviceMetadata/publishing/1.0/, id=http://busdox.org/transport/identifiers/1.0/, ds=http://www.w3.org/2000/09/xmldsig#, wsa=http://www.w3.org/2005/08/addressing}; ns2Prefix2={http://www.w3.org/2005/08/addressing=[wsa], http://www.w3.org/2000/09/xmldsig#=[ds], http://busdox.org/serviceMetadata/publishing/1.0/=[smp], http://busdox.org/transport/identifiers/1.0/=[id]}]: java.lang.IllegalArgumentException -- Failed to set JAXB property 'com.sun.xml.bind.namespacePrefixMapper' to [JAXBNamespacePrefixMapper@0x15d5378d: NamespaceContext=[MapBasedNamespaceContext@0x270e995f: prefix2NS={smp=http://busdox.org/serviceMetadata/publishing/1.0/, id=http://busdox.org/transport/identifiers/1.0/, ds=http://www.w3.org/2000/09/xmldsig#, wsa=http://www.w3.org/2005/08/addressing}; ns2Prefix2={http://www.w3.org/2005/08/addressing=[wsa], http://www.w3.org/2000/09/xmldsig#=[ds], http://busdox.org/serviceMetadata/publishing/1.0/=[smp], http://busdox.org/transport/identifiers/1.0/=[id]}]]`

(fwiw, this is in Wildfly v. 23.0.2)

Thanks for building an excellent application.

**Comment by phax** [comment]
Hi @OysteinLq  as I am not a Wildfly expert, I can only provide a guess.
Somebody else already provided a "helper" file called `WEB-INF/jboss-deployment-structure.xml` that I include in my deployments. It contains module dependencies, and excludes stuff that I think could cause your issues:
See https://github.com/phax/phoss-smp/blob/master/phoss-smp-webapp-xml/src/main/webapp/WEB-INF/jboss-deployment-structure.xml for an example (independent of the backend)
So maybe you need to "activate" this file or something similar???

Additionally. I am wondering why Java 12, as only 11 and 17 are long term support (LTS) versions.... But I assume you know what you are doing ;-)

**Reply by OysteinLq** [reply]
Thanks for the response -- the jboss-deployment-structure.xml file is placed in such a way that Wildfly-users don't need to do anything special to activate it, but we'll look into whether or not they might need altering now that JAXB's been moved out of Java. If so, we'll see if we can do a PR.

I won't argue whether or not I know what I'm doing, but I did mean Java 11, not 12, so I'll change the title

**Reply by phax** [reply]
Okay thanks for the heads up. To be honest I can't recall either the Wildfly version nor the Java version used by the guys providing the patch. If I don't forget, I'll try to figure it out
